### DISCUSSION TITLE
make booter works when typescript is configured to generate ECMA Scripts

### DISCUSSION THREAD

**Discussion by ecurien** [loopbackio/loopback-next#8753] [Ideas]
in booter-utils.ts there is a dynamic require on files from external code with loadClassesFromFiles.
It would be great to make this method async and use dynamic import that works with commonjs and module.

actually error is:
Error [ERR_REQUIRE_ESM]: require() of ES Module ...
Instead change the require of auth.controller.auth.js in ...\node_modules\@loopback\boot\dist\booters\booter-utils.js to a dynamic import() which is available in all CommonJS modules.

I  propose:
https://github.com/loopbackio/loopback-next/pull/8754

**Comment by achrinza** [comment]
Thanks for opening the PR and discussion. For reference, there's been some prior discussion here: https://github.com/loopbackio/loopback-next/discussions/8555
